File: OnlineTaksiJetpack/app/src/main/java/com/example/onlinetaksijetpack/data/remote/model/RideModels.kt
package com.example.onlinetaksijetpack.data.remote.model

data class CreateRideRequest(
    val pickup_lat: Double,
    val pickup_lng: Double,
    val pickup_address: String,
    val dropoff_lat: Double,
    val dropoff_lng: Double,
    val dropoff_address: String
)

data class RideResponse(
    val id: Int,
    val customer_id: Int,
    val assigned_driver_id: Int?,
    val pickup_lat: Double,
    val pickup_lng: Double,
    val pickup_address: String,
    val dropoff_lat: Double,
    val dropoff_lng: Double,
    val dropoff_address: String,
    val status: String,
    val estimated_fare: Double?,
    val final_fare: Double?,
    val cancel_reason: String?
)

File: online_taksi_react_native/src/models/rideModels.ts
export type CreateRideRequest = {
    pickup_lat: number;
    pickup_lng: number;
    pickup_address: string;
    dropoff_lat: number;
    dropoff_lng: number;
    dropoff_address: string;
};

export type RideResponse = {
    id: number;
    customer_id: number;
    assigned_driver_id?: number | null;
    pickup_lat: number;
    pickup_lng: number;
    pickup_address: string;
    dropoff_lat: number;
    dropoff_lng: number;
    dropoff_address: string;
    status: string;
    estimated_fare?: number | null;
    final_fare?: number | null;
    cancel_reason?: string | null;
};